### PR TITLE
Support for a custom base url

### PR DESCRIPTION
**Pull request by freeformz**
Useful for https://www.hipchat.com/server

### CODE REVIEW

File: hipchat.go [hunk 1]
```diff
@@ -77,6 +77,14 @@ type ErrorResponse struct {
 
 type Client struct {
 	AuthToken string
+	BaseURL   string
+}
+
+// NewClient returns a Client setup with the proper authToken that points to
+// the Hipchat service
+// Override BaseURL before using the Client to set a custom host
```
Comment (andybons) [inline hipchat.go]:
```
// NewClient allocates and returns a Client with the given authToken.
// By default, the client will use the publicly available HipChat servers.
// For internal or custom servers, set the BaseURL field of the Client.
```

File: hipchat.go [hunk 2]
```diff
@@ -77,6 +77,14 @@ type ErrorResponse struct {
 
 type Client struct {
 	AuthToken string
+	BaseURL   string
```
Comment (andybons) [inline hipchat.go]: change the name of the `baseURL` const to `defaultBaseURL` in lieu of this change.

File: README.md [hunk 3]
```diff
@@ -38,6 +38,14 @@ func main() {
...
+Setting a custom HipChat Server:
+```go
+c := hipchat.NewClient("<AUTH TOKEN>")
+c.BaseURL = "https://your.host.name/v1"
+...
+```
+
 Contributors
 ------------
 + Akshay Shah ([@akshayjshah](https://github.com/akshayjshah))
```
Comment (andybons) [inline README.md]: Please add yourself as a contributor :)

**Comment by andybons** [comment]
Thanks for contributing! This is great :)

Few minor changes, but overall looks good to me. Once you’ve updated I’ll merge.

**Comment by freeformz** [comment]
@andybons I think I covered everything.

**Pull request merged**